src/Interpreters/Context.cpp

```diff
@@ -1130,6 +1131,39 @@ String Context::getFilesystemCachesPath() const
     return shared->filesystem_caches_path;
 }
 
+std::shared_ptr<IDisk> Context::getDatabaseDisk() const
+{
+    {
+        SharedLockGuard lock(shared->mutex);
+        if (shared->db_disk)
+            return shared->db_disk;
+    }
+
+    auto target_db_disk = [&]() -> std::shared_ptr<IDisk>
```

Comment:
Would it be better if we add a separate mutex to create target_db_disk and avoid several threads creating it?

Reply:
In my opinion, it is not necessary. `getDatabaseDisk()` is called first time early during the initialization. Even if multiple threads try to get `target_db_disk`, only the first one will initialize the disks as there is another mutex in `getDiskMap()`.

Reply:
Indeed, add this to a comment here?